### DISCUSSION TITLE
resource utilization

### DISCUSSION THREAD

**Discussion by alexa22013** [emqx/emqx-operator#597] [General]
Hey as per this blog https://www.emqx.com/en/blog/how-emqx-5-0-achieves-100-million-mqtt-connections#:~:text=Core%20nodes%20require%20a%20large%20amount%20of%20memory%2C%20and%20the%20CPU%20consumption%20is%20low%20when%20no%20connections%20are%20undertaken%3B%20the%20hardware%20configuration%20of%20Replicant%20nodes%20is%20consistent%20with%20v4.x%2C%20and%20its%20memory%20requirements%20can%20be%20estimated%20according%20to%20the%20connection%20and%20throughput%20configuration.

emqx core uses more cpu and memory than replicant, but when I deployed a cluster witth 1 core and 2 replicants I observed that the replicant used almost double the cpu and memory of core node

any specific reason?

**Comment by hjianbo** [comment]
> Core nodes require a large amount of memory, and the CPU consumption is low when no connections are undertaken; the hardware configuration of Replicant nodes is consistent with v4.x, and its memory requirements can be estimated according to the connection and throughput configuration.

In short, the implication of this paragraph is that **the memory requirements of core nodes are obviously higher than the CPU requirements when core nodes do not take connections.**

However, the relationship of CPU and memory between the replicant node and the core node is uncertain, depending on what kind of scenario is being run on the replicant node.
